crypto/fipsmodule/service_indicator/service_indicator_test.cc

```diff
@@ -508,6 +508,100 @@ static const uint8_t kTLSOutput_sha512[32] = {
+    // Internal IV usage of AES-GCM is approved.
+    { EVP_aead_aes_128_gcm_randnonce(), kAESKey, 16, nullptr, 0, AWSLC_APPROVED },
+    { EVP_aead_aes_256_gcm_randnonce(), kAESKey_256, 32, nullptr, 0, AWSLC_APPROVED },
+    // External IV usage of AES-GCM is not approved unless used within a TLS
+    // context.
+    { EVP_aead_aes_128_gcm(), kAESKey, 16, kAESGCMCiphertext_128, 64, AWSLC_NOT_APPROVED },
+    { EVP_aead_aes_192_gcm(), kAESKey_192, 24, kAESGCMCiphertext_192, 64, AWSLC_NOT_APPROVED },
+    { EVP_aead_aes_256_gcm(), kAESKey_256, 32, kAESGCMCiphertext_256, 64, AWSLC_NOT_APPROVED },
+    // External IV usage of AEAD AES-GCM APIs specific for TLS is approved.
+    { EVP_aead_aes_128_gcm_tls12(), kAESKey, 16, kAESGCMCiphertext_128, 64, AWSLC_APPROVED },
+    { EVP_aead_aes_256_gcm_tls12(), kAESKey_256, 32, kAESGCMCiphertext_256, 64, AWSLC_APPROVED },
+    { EVP_aead_aes_128_gcm_tls13(), kAESKey, 16, kAESGCMCiphertext_128, 64, AWSLC_APPROVED },
+    { EVP_aead_aes_256_gcm_tls13(), kAESKey_256, 32, kAESGCMCiphertext_256, 64, AWSLC_APPROVED },
+    // 128 bit keys with 32 bit tag lengths are approved for AES-CCM.
+    { EVP_aead_aes_128_ccm_bluetooth(), kAESKey, 16, kAESCCMCiphertext, 64, AWSLC_APPROVED },
+};
+
+class AEAD_ServiceIndicatorTest : public testing::TestWithParam<AEADTestVector> {};
+
+INSTANTIATE_TEST_SUITE_P(All, AEAD_ServiceIndicatorTest, testing::ValuesIn(nAEADTestVectors));
+
+TEST_P(AEAD_ServiceIndicatorTest, EVP_AEAD) {
+  const AEADTestVector &aeadTestVector = GetParam();
+
+  int approved = AWSLC_NOT_APPROVED;
+
+  bssl::ScopedEVP_AEAD_CTX aead_ctx;
+  std::vector<uint8_t> plaintext(kPlaintext, kPlaintext + sizeof(kPlaintext));
+  std::vector<uint8_t> nonce(EVP_AEAD_MAX_NONCE_LENGTH, 0);
+  std::vector<uint8_t> encrypt_output(256);
+  std::vector<uint8_t> decrypt_output(256);
+  size_t out_len;
+
+  // Test running the EVP_AEAD_CTX interfaces one by one directly, and check
+  // |EVP_AEAD_CTX_seal| and |EVP_AEAD_CTX_open| for approval at the end.
+  // |EVP_AEAD_CTX_init| should not be approved because the function does not
+  // indicate that a service has been fully completed yet.
+  CALL_SERVICE_AND_CHECK_APPROVED(approved, ASSERT_TRUE(EVP_AEAD_CTX_init(aead_ctx.get(), aeadTestVector.cipher,
+                                aeadTestVector.key, aeadTestVector.key_length, 0, nullptr)));
+  ASSERT_EQ(approved, AWSLC_NOT_APPROVED);
+  CALL_SERVICE_AND_CHECK_APPROVED(approved, ASSERT_TRUE(EVP_AEAD_CTX_seal(aead_ctx.get(),
+      encrypt_output.data(), &out_len, encrypt_output.size(), nonce.data(), EVP_AEAD_nonce_length(aeadTestVector.cipher),
+      plaintext.data(), plaintext.size(), nullptr, 0)));
+  ASSERT_EQ(approved, aeadTestVector.expect_approved);
+  encrypt_output.resize(out_len);
+  if(aeadTestVector.expected_ciphertext) {
+      ASSERT_TRUE(check_test(aeadTestVector.expected_ciphertext, encrypt_output.data(),
+                           aeadTestVector.cipher_text_length, "AES-GCM Encryption"));
```

Comment:
It will say "AES-GCM" for CCM as well. You may want to put a tailored Log message in the parameter set.